source/dagger/gameplay/team_game/controls.cpp

```diff
@@ -0,0 +1,74 @@
+#include "controls.h"
+#include "mage.h"
+#include "golem.h"
+#include "towers.h"
+
+#include "core/input/inputs.h"
+#include "core/engine.h"
+
+using namespace ancient_defenders;
+
+Float32 ancient_defenders::ControlData::timeout = 0.0f;
```

Comment:
it can be a simple struct member inside PlayerControlsSystem. No need to make it static.

maxTimeout  can be a constexpr value.